ddterm/shell/windowmatch.js

```diff
@@ -43,9 +46,15 @@ export const WindowMatchGeneric = GObject.registerClass({
             }
         });
 
-        Meta.get_window_actors(this.display).forEach(actor => {
-            this._watch_window(actor.meta_window);
-        });
+        if (shellVersion < 48) {
+            Meta.get_window_actors(this.display).forEach(actor => {
+                this._watch_window(actor.meta_window);
+            });
+        } else {
+            global.compositor.get_window_actors(this.display).forEach(actor => {
+                this._watch_window(actor.meta_window);
+            });
```

Comment:
There's no need to have two copies of the same `forEach()`. You could store the return value of the correct `.get_window_actors()` in a variable, and then call `.forEach()` on it - only once, outside of `if`.